src/Microsoft.AspNetCore.Sockets/Internal/AuthorizeHelper.cs

```diff
@@ -12,53 +13,62 @@ namespace Microsoft.AspNetCore.Sockets.Internal
 {
     public static class AuthorizeHelper
     {
-        public static async Task<bool> AuthorizeAsync(HttpContext context, AuthorizationPolicy policy)
+        public static async Task<bool> AuthorizeAsync(HttpContext context, IList<string> policies)
         {
-            if (policy != null)
+            if (policies.Count == 0)
             {
-                if (policy.AuthenticationSchemes != null && policy.AuthenticationSchemes.Count > 0)
-                {
-                    ClaimsPrincipal newPrincipal = null;
-                    foreach (var scheme in policy.AuthenticationSchemes)
-                    {
-                        var result = await context.Authentication.AuthenticateAsync(scheme);
-                        if (result != null)
-                        {
-                            newPrincipal = SecurityHelper.MergeUserPrincipal(newPrincipal, result);
-                        }
-                    }
-
-                    if (newPrincipal == null)
-                    {
-                        newPrincipal = new ClaimsPrincipal(new ClaimsIdentity());
-                    }
+                return true;
+            }
 
-                    context.User = newPrincipal;
-                }
+            var policyProvider = context.RequestServices.GetRequiredService<IAuthorizationPolicyProvider>();
+            var authorizeAttributes = new List<AuthorizeAttribute>();
```

Comment:
Yeah it is odd, https://github.com/aspnet/Security/blob/22d2fe99c6fd9806b36025399a217a3a8b4e50f4/src/Microsoft.AspNetCore.Authorization/AuthorizationPolicy.cs#L111 the only thing that implements IAuthorizeData is AuthorizeAttribute.